### PR TITLE
Issue#1421 main information

### PR DESCRIPTION
**Pull request by Hugo-Hontans**
- [x] Movie Form Display for tunnel-summary and dashboard title-view
- [x] Create summary tunnel component
- [x] Create different components in movie/form/summary/* to display the main part

### CODE REVIEW

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<tunnel-page>
+  <h4 title>END</h4>
+  <h1 subtitle>Summary of Title & Licensed Rights</h1>
+  <p description class="mat-body-1">
+    Please verify that all the information you have entered is correct before submitting to Archipel Content for validation.
+  </p>
+
+  <section tunnel-form>
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.component.html]: Use <ng-container tunnel-form> instead and set card to width 100%

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.module.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { TunnelSummaryComponent } from './summary.component';
+import { MovieDisplayModule } from '@blockframes/movie/moviedisplay/display.module';
+
+@NgModule({
+  declarations: [TunnelSummaryComponent],
+  imports: [
+    CommonModule,
+    FlexLayoutModule,
+    TunnelPageModule,
+    MovieDisplayModule,
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.module.ts]: You shouldn't put everything in one module. Split in several modules

File: libs/movie/src/lib/movie/display/main/main.component.html [hunk 3]
```diff
@@ -1,81 +1,105 @@
...
-    <ng-container *ngIf="data.shortSynopsis">
-      <mat-card-title>Short Synopsis</mat-card-title>
-      <mat-card-subtitle>
-        <p>{{ data.shortSynopsis }}</p>
-      </mat-card-subtitle>
+<h2>Title Information</h2>
+<mat-card>
+  <div class="information" fxLayout="row" fxLayoutAlign="start center">
+    <h3>Main Information</h3>
+    <a mat-icon-button routerLink="../../../../../../main"><mat-icon svgIcon="edit"></mat-icon></a>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/main/main.component.html]: What is this route ???

File: libs/movie/src/lib/movie/display/main/main.component.html [hunk 4]
```diff
@@ -1,81 +1,105 @@
...
+  </div>
+
+  <mat-card-content>
+    <p>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/main/main.component.html]: It looks like a big list for me, do something like : 
```html
<ul class="mat-body-1">
  <ng-container *ngFor="let field of fields | keyvalue">
    <li><strong>{{ field.value }}</strong> {{ movie[field.key] | missing }}</li>
  </ng-container>
</ul>
```

Comment (Hugo-Hontans) [reply libs/movie/src/lib/movie/display/main/main.component.html]: I'm not sure to understand:
you would like to iterate on each control of the form but the text to display is different from the key of the controls. Moreover I have to control the different fields that I display.

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.component.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<tunnel-page>
+  <h4 title>END</h4>
+  <h1 subtitle>Summary of Title & Licensed Rights</h1>
+  <p description class="mat-body-1">
+    Please verify that all the information you have entered is correct before submitting to Archipel Content for validation.
+  </p>
+
+  <mat-card tunnel-form>
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/summary/summary.component.html]: I think it's several card : 
```html
<ng-container tunnel-form>
  <mat-card>
    <mat-card-header>
      <mat-card-title>Main</...>
    </mat-card-header>
    <mat-card-content>
      <movie-summary-main [main]="form.main"></movie-summary-main>
     </...>
  </mat-card>
</ng-container>

File: libs/movie/src/lib/movie/form/main/main.form.ts [hunk 6]
```diff
@@ -125,6 +125,26 @@ export class MovieMainForm extends FormEntity<MovieMainControl>{
     super(createMovieMainControls(main));
   }
 
+  public get genres() {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/main/main.form.ts]: Honestly we all use the `get(key)` in the other component. I'm fine with that, but it's not needed.

File: libs/movie/src/lib/movie/form/summary/country/country.component.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<ng-container *ngFor="let country of originCountries.controls; let i = index">
+  <ul>
+    <li class="mat-body-2">Countries of Origin #{{ i + 1 }}</li>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/country/country.component.html]: use `h5` on top of the ul. This looks like a title to me.

File: libs/movie/src/lib/movie/form/summary/sales-cast/sales-cast.component.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<ul>
+  <li>
+    <strong>Principal Cast </strong>
+    <ng-container *ngFor="let cast of salesCast.cast.controls; let isLast = last">
+      <ng-container *ngIf="cast.value === 'lead-role'">
+        {{ cast.get('firstName') | missing }} {{ cast.get('lastName') | missing }}{{ isLast ? '' : ', ' }}
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/sales-cast/sales-cast.component.html]: What about a "." if it's last ?

File: libs/utils/src/lib/pipes/missing.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+@Pipe({
+  name: 'missing'
+})
+export class MissingPipe implements PipeTransform {
+
+  transform(control: FormControl) {
+    return control.hasError('required') ? 'Mandatory' : control.value ? control.value : 'Missing';
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/missing.pipe.ts]: So there is no way to transform it into an HTML Element ?
I think that at some point we'll need to transform that into a directive to be able to create link to the right place in the form. Maybe you can create an issue for that.

Comment (Hugo-Hontans) [reply libs/utils/src/lib/pipes/missing.pipe.ts]: I have the impression that we cannot return an html element (in which we could add style).
I left that aside to go ahead.
But there is certainly a solution to change the color with the pipe

File: libs/movie/src/lib/movie/form/summary/sales-cast/sales-cast.component.html (hunk 8 above)
Comment (Hugo-Hontans) [reply libs/movie/src/lib/movie/form/summary/sales-cast/sales-cast.component.html]: There is no "." in the wireframes but I will find out.
I will modify later if necessary

**Pull request merged**